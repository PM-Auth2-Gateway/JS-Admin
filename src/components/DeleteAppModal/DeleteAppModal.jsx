import React, { useState } from 'react';
import { Button, Modal } from 'react-bootstrap';

const DeleteAppModal = ({ onDelete }) => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const onSubmit = (values) => {
    onDelete();
    handleClose();
  };

  return (
    <React.Fragment>
      <Button data-test={'open-modal'} variant='danger' onClick={handleShow}>
        Delete
      </Button>

      <Modal show={show} onHide={handleClose} centered>
        <Modal.Header closeButton>
          <Modal.Title>Delete Application</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          Are you sure you want to delete this application?
        </Modal.Body>

        <Modal.Footer>
          <Button
            data-test={'close-modal'}
            variant='outline-secondary'
            onClick={handleClose}
          >
            Cancel
          </Button>
          <Button
            data-test={'submit-modal'}
            variant='primary'
            onClick={onSubmit}
          >
            Delete
          </Button>
        </Modal.Footer>
      </Modal>
    </React.Fragment>
  );
};

export default DeleteAppModal;
