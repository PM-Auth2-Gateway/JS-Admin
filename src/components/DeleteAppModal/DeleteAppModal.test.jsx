import React, { useState as useStateMock } from 'react';
import { shallow } from 'enzyme';

import DeleteAppModal from './DeleteAppModal';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('DeleteAppModal Component', () => {
  const setState = jest.fn();
  const onDelete = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setState]);
  });

  it('should render DeleteAppModal Component', function () {
    const component = shallow(<DeleteAppModal />);

    expect(component).toMatchSnapshot();
  });

  it('should open modal', function () {
    const component = shallow(<DeleteAppModal />);

    const openModalBtn = component.find(`[data-test="open-modal"]`);

    openModalBtn.simulate('click');
  });

  it('should close modal', function () {
    const component = shallow(<DeleteAppModal />);

    const closeModalBtn = component.find(`[data-test="close-modal"]`);

    closeModalBtn.simulate('click');
  });

  it('should submit modal', function () {
    const component = shallow(<DeleteAppModal onDelete={onDelete} />);

    const submitModalBtn = component.find(`[data-test="submit-modal"]`);

    submitModalBtn.simulate('click');
  });
});
