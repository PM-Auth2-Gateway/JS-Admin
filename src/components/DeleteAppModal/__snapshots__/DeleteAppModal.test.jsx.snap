// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAppModal Component should render DeleteAppModal Component 1`] = `
<Fragment>
  <Button
    active={false}
    data-test="open-modal"
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    Delete
  </Button>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle>
        Delete Application
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      Are you sure you want to delete this application?
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        data-test="close-modal"
        disabled={false}
        onClick={[Function]}
        variant="outline-secondary"
      >
        Cancel
      </Button>
      <Button
        active={false}
        data-test="submit-modal"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Delete
      </Button>
    </ModalFooter>
  </Modal>
</Fragment>
`;
